scipy/sparse/_matrix_io.py

```diff
@@ -72,7 +69,9 @@ def save_npz(file, matrix, compressed=True):
         arrays_dict.update(row=matrix.row, col=matrix.col)
     else:
         raise NotImplementedError('Save is not implemented for sparse matrix of format {}.'.format(matrix.format))
-
+    arrays_dict = dict(format=matrix.format.encode('ascii'),
```

Comment:
This should update arrays_dict, not overwrite it.

Reply:
Thanks. Done.